README.md

```diff
@@ -1 +1,430 @@
+
+* With HTTPS:
+```bash
+git submodule add --name STM32RF24 https://github.com/ThundeRatz/STM32RF24.git lib/STM32RF24
+```
+
+* With SSH:
+```bash
+git submodule add --name STM32RF24 git@github.com:ThundeRatz/STM32RF24.git lib/STM32RF24
+```
+
+### ✅ Initializing a existing submodule
+
+When cloning a repository that already has submodules, it is necessary to clone the repositories of that submodule. This can be done in two ways, by cloning together with the project repository or after you have already cloned.
+
+Example:
+
+To clone together, run the following command, switching to the repository link of your project:
+
+```bash
+git clone --recurse-submodules git@github.com:ThundeRatz/STM32ProjectTemplate.git
+```
+
+To clone having already cloned the project repository, within it, you should do:
+
+```bash
+git submodule update --init
+```
+
+## 📁 Repository Structure
+
+The repository contains the following folders:
+
+- **docs/** → Documentation files
+- **inc/** → Header files
+- **src/** → Source files
+
+At the root of the repository, in addition to the files containing the code of conduct, contribution guidelines, README and license, there is the `sources.mk` file, which is responsible for making it possible for the library files to be found when compiling the code. There is also a `Doxyfile` to generate the documentation. Another relevant file is `uncrustify.cfg` which is used to format the files.
+
+## 📝 Documentation
+
+This library is documented using Doxygen. To generate the documentation, run the following command at the root of your repository:
+
+```bash
+doxygen
+```
+
+The settings are in the file [Doxyfile](./Doxyfile).
+
+## 🎨 Formatting
+
+Besides a good documentation, it is necessary that the code is always well formatted, which facilitates its understanding. For that, `uncrustify` was used. With `uncrustify` installed, to format a file that has changed, run the following command:
+
+```bash
+uncrustify -c uncrustify.cfg --replace --no-backup path_to_file/file_name
+```
+
+## 🏗️ Code Structure
+
+The code is structured as follows:
+
+- `nrf24l01_registers.h` → types and constants related to the module registers.
+- `rf24_platform.c/.h` → lower-level types and functions that use HAL.
+- `rf24.c/.h` → highest level types and functions for user use.
+- `rf24_debug.c/.h` → useful functions to validate the module's operation.
+
+## 🔌 Hardware Configuration
+
+### 🤔 Getting to know the module
+
+To configure your hardware, you must first analyze the pinout of the nRF24L01 module, as shown below:
+
+![nRF24L01 Module's Pinout](./assets/img/nrf24l01_pinout.png)
+
+The module uses SPI (Serial Protocol Interface) to communicate with the microcontroller (to learn more about SPI, I recommend this article [here](https://learn.sparkfun.com/tutorials/serial-peripheral-interface-spi/all) from Sparkfun), so four pins are required for this communication, SCK, MISO, MOSI and CSN. The CSN is a GPIO pin, while the others are dedicated pins.
```

Comment:
we recommend?
Não sei, acho estranho readme na primeira pessoa hahahhah

Reply:
Justo, acho melhor mesmo hsuahsauhsuashauh.